src/com/amazon/corretto/crypto/provider/EcUtils.java

```diff
@@ -108,12 +112,27 @@ private String normalizeName(final String name) {
                 }
             } else if (name.startsWith("X9.62 ")) {
                 return name.substring(6);
+            } else if (name.equals("secp256r1")) {
+                return "prime256v1";
             } else {
                 return name;
             }
         }
     };
 
+    private static final ConcurrentHashMap<ByteBuffer, String> EC_NAME_BY_ENCODED = new ConcurrentHashMap<>();
+    private static final ConcurrentHashMap<EllipticCurve, String> EC_NAME_BY_CURVE = new ConcurrentHashMap<>();
+    private static final ConcurrentHashMap<Integer, String> EC_NAME_BY_KEY_SIZE = new ConcurrentHashMap<>();
+    static {
+        for (String name : getCurveNames()) {
+            getSpecByName(name);                    // warm cache keyed by name, must come before warming by OID
+            getSpecByName(getOidFromName(name));    // warm cache keyed by OID
```

Comment:
Comments could be a bit clearer. Is this just a performance optimization for the static initializer and has no impact after the class is loaded?

Reply:
> Is this just a performance optimization for the static initializer and has no impact after the class is 

it's a performance enhancement for runtime post-initialization. we pay the (small) JNI cost a few times during static initialization, then every cache hit after that is just a HashMap get(). this is safe because the set of supported curves is invariant after loading AWS-LC.

i'll expand the comments a bit.